src/vtbackend/JumpHistory.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <vtbackend/primitives.h>
+
+#include <vector>
+
+namespace vtbackend
+{
+
+struct JumpHistory
```

Comment:
if we have private data/methods, it should be a `class` instead (ref: CppCoreGuidelines IIRC)